### PR TITLE
Add AutoComplete

### PR DESCRIPTION
**Pull request by DRSchlaubi**
<https://github.com/discord/discord-api-docs/pull/3996>

- Add support for AutoComplete options
- Update Core representation

### CODE REVIEW

File: rest/src/main/kotlin/builder/interaction/OptionsBuilder.kt [hunk 1]
```diff
@@ -27,12 +26,26 @@ sealed class OptionsBuilder(
...
+    internal var _autocomplete: OptionalBoolean = OptionalBoolean.Missing
+
+    /**
+     * Setting this to `true` allows you to dynamically respond with your choices, depending on the user input.
+     *
+     * **Warning:** This disables all input validation, users can submit values before responding to the AutoComplete request
+     *
+     * **Note:** If you set this to `true` you can't add any other choice
+     *
+     * Core users can check the `AutoCompleteInteractionCreateEvent` and `AutoCompleteInteraction` classes.
```
[the remark this answers is not in the document]
Comment (HopeBaron) [reply rest/src/main/kotlin/builder/interaction/OptionsBuilder.kt:38]: use mark down to link to the mentioned classes

**Review by HopeBaron — CHANGES_REQUESTED** [review]

**Comment by DRSchlaubi** [comment]
@HopeBaron Those classes are in the core module, the file above is rest, so these links don't work
No idea why but couldn't reply to your convo above directly

**Comment by HopeBaron** [comment]
There is no need for such documentation when you can't refer to it. rest module is unaware of events.

This sort of documentation may become invalid since refactoring may now work on those so they will become lies if things change. And Kord has already enough of those kinds of documentation that we should be working on already. 
Remove them for now also I'd like you to integrate warnings and Notes without bold headers.

**Comment by HopeBaron** [comment]
why "MessageResponding"
I guess you mixed it up with `InteractionResponse`?

**Comment by DRSchlaubi** [comment]
Because these interactions respond with a Message unlike AutoComplete which respond with a Choices object.

I think it was @BartArys who suggested MessageResponding

**Comment by HopeBaron** [comment]
not all commands should respond with a message. it may be a command with no message, and chatinput now implements only `Interaction` instead of `ApplicationCommandInteraction`
I don't get why this happened...

They are all application commands, it's just that AutoComplete responds diferently, right?

**Comment by DRSchlaubi** [comment]
> and chatinput now implements only Interaction instead of ApplicationCommandInteraction
I don't get why this happened...

Okay that was unintended

> They are all application commands, it's just that AutoComplete responds diferently, right?

yes this is why MessageRespondingInteraction is everything but autocomplete, but I am open for better names

**Pull request merged**